### PR TITLE
[WIP] Set keepalive on Unix Socket

### PR DESCRIPTION
**Pull request by axic**
Fixes the connection closed issue from https://github.com/ethereum/solidity/issues/3825

### CODE REVIEW

File: libweb3jsonrpc/UnixSocketServer.cpp [hunk 1]
```diff
@@ -86,6 +86,15 @@ bool UnixDomainSocketServer::StartListening()
 		::bind(m_socket, reinterpret_cast<sockaddr*>(&m_address), sizeof(sockaddr_un));
 		fs::permissions(m_path, fs::owner_read | fs::owner_write);
 		listen(m_socket, 128);
+
+		int keepalive = 1;
+		if (setsockopt(m_socket, SOL_SOCKET, SO_KEEPALIVE, &keepalive, sizeof(keepalive) < 0)
```
Comment (gumb0) [inline libweb3jsonrpc/UnixSocketServer.cpp]: syntax error
```suggestion
		if (setsockopt(m_socket, SOL_SOCKET, SO_KEEPALIVE, &keepalive, sizeof(keepalive)) < 0)
```

File: libweb3jsonrpc/UnixSocketServer.cpp [hunk 2]
```diff
@@ -141,7 +150,7 @@ size_t UnixDomainSocketServer::Write(int _connection, string const& _data)
 
 size_t UnixDomainSocketServer::Read(int _connection, void* _data, size_t _size)
 {
-	ssize_t r = read(_connection, _data, _size);
+	ssize_t r = recv(_connection, _data, _size, MSG_NOSIGNAL);
```
Comment (chfast) [inline libweb3jsonrpc/UnixSocketServer.cpp]: What does it do?

**Comment by axic** [comment]
Hold on a sec, I have more fixes to this PR.

**Comment by chfast** [comment]
I'm newbie in context of socket programming. Is this still a problem?

Some time ago I tried to use the unix socket server from libjson-rpc-cpp #4565 but it didn't work at all. There is not a lot of changes there (https://github.com/cinemast/libjson-rpc-cpp/commits/master/src/jsonrpccpp/server/connectors/unixdomainsocketserver.cpp) but we can try again. Maybe they can setup it better than we are doing it currently.

**Comment by axic** [comment]
It is, but this PR isn't finished. Please don't merge.

**Comment by axic** [comment]
I will not have time to ever finish this PR and it is not an issue anymore in Solidity. The problem however persists if anyone wants to fix it.

**Pull request closed without merging**